File: backend/src/main/java/com/backend/ord/seeders/entitiesSeeders/UserSeeder.java
package com.backend.ord.seeders.entitiesSeeders;

import com.backend.ord.domain.entities.User;
import org.springframework.stereotype.Component;
import com.github.javafaker.Faker;

@Component
public class UserSeeder {
    private UserSeeder() {
    }

    private final static Faker faker = Faker.instance();

    public static User createUser() {
        return User.builder()
                .name(faker.name().fullName())
                .email(faker.internet().emailAddress())
                .build();
    }

    public static User createUser(String name, String email) {
        return User.builder()
                .name(name)
                .email(email)
                .build();
    }

    public static User createUser(String email) {
        return User.builder()
                .name(faker.name().fullName())
                .email(email)
                .build();
    }

}

File: src/test/kotlin/com/backend/ord/seeders/entities/UserSeeder.kt
package com.backend.ord.seeders.entities

import com.backend.ord.core.user.UserRepository
import com.backend.ord.core.user.model.UserEntity
import com.backend.ord.seeders.entities.bases.SeederInterface
import com.backend.ord.seeders.factories.UserFactory
import org.springframework.stereotype.Component

@Component
class UserSeeder(
    private val userRepository: UserRepository,
    private val userMockFactory: UserFactory
) : SeederInterface<UserEntity> {
    override fun seedOneEntity(data: UserEntity?): UserEntity {
        return userRepository.save(data ?: userMockFactory.mockEntity())
    }

    fun insertRowWithCredentials(email: String, password: String): UserEntity {
        return userRepository.save(
            userMockFactory.mockEntity(
                email = email,
                password = password
            )
        )
    }

    override fun deleteAll() {
        userRepository.deleteAll()
    }
}
